packages/common/src/util.js

```diff
@@ -63,3 +82,137 @@ export function normalizeOauthConfig(configuration) {
 
   return configuration;
 }
+
+const separateUrl = fullUrl => {
+  const url = new URL(fullUrl);
+  return {
+    baseUrl: url.origin,
+    path: url.pathname,
+  };
+};
+
+/**
+ * The `defaultOptions` constant is an object that defines default values for various options that can
+ * be passed to the `request` function. These options include
+ * `headers`,
+ * `query`,
+ * `timeout`,
+ * `tls`,
+ * `body`, and
+ * `errors`.
+ */
+const defaultOptions = {
+  timeout: 300, // Default to 300 seconds
+  headers: {},
+  query: undefined,
+  body: undefined,
+  errors: {},
+  tls: {},
+};
+
+const assertOK = (response, errorMap, fullUrl) => {
+  const errMapMessage = errorMap[response.statusCode];
+
+  if (errMapMessage || response.statusCode >= 400) {
+    const defaultErrorMesssage = `Request to ${fullUrl} failed with status: ${response.statusCode}`;
+
+    const errMessage =
+      typeof errMapMessage === 'function'
+        ? errMapMessage(response)
+        : errMapMessage || defaultErrorMesssage;
+
+    const error = new Error(errMessage);
+    error.code = response.statusCode;
+    error.url = fullUrl;
+    throw error;
+  }
+};
+
+const isValidHttpUrl = fullUrl => {
+  try {
+    const { protocol } = new URL(fullUrl);
+    return protocol === 'http:' || protocol === 'https:';
+  } catch (_) {
+    return false;
+  }
+};
+
+/**
+ * The function `request` is an asynchronous function that sends HTTP requests and returns the response
```

Comment:
It is nice to see good, thorough documentation here  - but remember this function won't be exposed directly to job writers, it won't be documented on docs.openfn.org